contracts/PoolManager.sol

```diff
@@ -122,7 +122,9 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
         PoolId id = key.toId();
         uint24 protocolFees = _fetchProtocolFees(key);
         uint24 hookFees = _fetchHookFees(key);
-        tick = pools[id].initialize(sqrtPriceX96, protocolFees, hookFees);
+        uint24 dynamicFee = _fetchDynamicFee(key);
```

Comment:
why do we put these into locals when we never need them again? slightly cleaner imo to fetch them inside the initialize call below as it better expresses that they're only intended to be used there. also probably saves some stack operations